google/auth/impersonated_credentials.py

```diff
@@ -321,6 +321,47 @@ def with_quota_project(self, quota_project_id):
             iam_endpoint_override=self._iam_endpoint_override,
         )
 
+    @classmethod
+    def from_authorized_user_info(cls, source_credentials, info, scopes=None):
+        """Creates a Credentials instance from parsed authorized user info.
+
+        Args:
+            source_credentials (google.oauth2.credentials.Credentials): The source
+                credentials required to impersonate the given principal.
+            info (Mapping[str, str]): The authorized user info in Google
+                format.
+            scopes (Sequence[str]): Optional list of scopes to include in the
+                credentials.
+
+        Returns:
+            google.impersonated_credentials.Credentials: The constructed
+                credentials.
+
+        Raises:
+            ValueError: If the info is not in the expected format.
+        """
+        import re
+
+        keys_needed = {"delegates", "service_account_impersonation_url", "source_credentials"}
+        missing = keys_needed.difference(six.iterkeys(info))
+
+        if missing:
+            raise ValueError(
+                "Authorized user info was not in the expected format, missing "
+                "fields {}.".format(", ".join(missing))
+            )
+
+        iam_endpoint_parser = re.search(pattern=r"(?P<service_account>[a-zA-Z-0-9]+)@(?P<project_id>[a-zA-Z-0-9]+)",
+                                        string=info.get("service_account_impersonation_url"))
+
+        return cls(
+            source_credentials=source_credentials,
+            target_principal=iam_endpoint_parser.group("service_account"),
+            target_scopes=["https://www.googleapis.com/auth/cloud-platform"],
+            delegates=info.get("delegates"),
+            iam_endpoint_override=info.get("service_account_impersonation_url")
+        ), iam_endpoint_parser.group("project_id")
```

Comment:
do you need to return this? It does not align with the docstring.

Reply:
I returned the `project_id` to follow the same mindset of the other type of auth but I can try if it's actually required.

Reply:
I tried, and it worked without. I seemed to be slower though. I guess the `project_id` should be inferred in some way later:
https://github.com/googleapis/google-auth-library-python/blob/e756f08dc78616040ab8fbd7db20903137ccf0c7/google/auth/_default.py#L197-L198

Because I already parse `service_account_impersonation_url` from the ADC, I propose to return it as well.
(rough measure with `time.time()` gave 0.5s when returning the `project_id` vs 1.8s without)

Reply:
@liuchaoren any news on that?